tests/api_test.py

```diff
@@ -1093,6 +1094,19 @@ def vjp(x_tangent):
     b = np.dot(a + np.eye(a.shape[0]), real_x)
     print(gf(a, b))  # doesn't crash
 
+  def test_device_put_replicated(self):
+    x = onp.array([3, 1, 4, 1])
+    y = api.device_put_replicated(x)
+    expected_shape = (xb.device_count(), 4)
+    self.assertEqual(y.shape, (xb.device_count(), 4))
+    self.assertAllClose(y, onp.broadcast_to(x, expected_shape),
+                        check_dtypes=False)
+
+    self.assertIsInstance(y, ShardedDeviceArray)
+    bufs = y.device_buffers
+    expected_ids = list(range(xb.device_count()))
```

Comment:
This might be true on the platforms we test on, but we shouldn't make assumptions about device IDs. I would replace this with self.assertEqual(len(set(b.device().id for b in bufs)), xb.device_count())

Reply:
You're right! Done.